### DISCUSSION TITLE
Access to Clips and Notes

### DISCUSSION THREAD

**Discussion by drdavient** [mkb79/audible-cli#73] [Ideas]
I'm looking for a way to access (and export) the Notes and Audio clips I take when listening to audio books. Is this an API endpoint you are likely to be investigating in future? Its a desperately missing feature of Audible - the app access to notes/clips is very poor. 

Initial thoughts on what this might look like. 
- Notes / Clips Endpoint
    - Each Book has an associated collection of Clips each with a given timestamp.
        - NoteText : string
        - Timestamp : int
        - ClipLength : int

Do you have any plans to build something like this?

**Comment by mkb79** [comment]
I‘m not sure if Notes/Clipes are synced between devices. I doesn’t find any network traffic from the Audible iOS app which is Notes/Clipes specific. So I think the app will store them only locally? I know you can get Notes from Kindle app for whispersync titles.

**Reply by drdavient** [reply]
I make all my notes on Android, these notes are definitely synced to the cloud.  
Though it looks like the "Audio Clip" is reduced to being a timestamped bookmark only.


To see them you need to open the Audible Cloud Player  and click the three dots menu on the top right:
![brave_XN47v3s8Rm](https://user-images.githubusercontent.com/4351544/160159266-d1213fdf-4c46-4f1f-ba2e-28b902c1f3dd.png)
![brave_M6Sf3FsQo7](https://user-images.githubusercontent.com/4351544/160159274-d41b1d8f-9267-4a07-89a4-2f74c6fef99b.png)
![brave_PIm8vDlEmB](https://user-images.githubusercontent.com/4351544/160159285-4fe7c2e1-bcc6-45fa-81a4-8915f04b3398.png)
So That would give us the structure:


* BookmarkName : string
* BookmarkLocation : int
* Bookmark Date : int (?)
* Note : string

**Reply by mkb79** [reply]
Okay, I found out how to set and get notes.

You can try this to get them:

```python
import httpx
from audible import Authenticator


auth = Authenticator.from_file(...)
asin = "..."

url = f"https://cde-ta-g7g.amazon.com/FionaCDEServiceEngine/sidecar"
params = {
    "type": "AUDI",
    "key": asin
}
with httpx.Client(auth=auth, params=params) as session:
    r = session.get(url)
    sidecar = r.json()


print(sidecar)
```

To set them is not so easy. You have to post a xml 
 body to the url above.

**Reply by mkb79** [reply]
[Here](https://github.com/mkb79/Audible/issues/43) you can read a bit more about your question!

**Reply by drdavient** [reply]
Thanks!  That should do the trick :)
